src/lib/layers/layers-counter.component.ts

```diff
@@ -8,23 +8,34 @@ import { FaLayersComponent } from './layers.component';
   selector: 'fa-layers-counter',
   template: '',
   host: {
-    class: 'ng-fa-layers-counter',
+    class: 'fa-layers-counter',
   },
 })
-export class FaLayersCounterComponent implements OnChanges {
+export class FaLayersCounterComponent implements OnChanges, OnInit {
   @Input() content: string;
   @Input() title?: string;
   @Input() styles?: Styles;
   @Input() classes?: string[] = [];
+  @Input() position?: string;
 
   @HostBinding('innerHTML') renderedHTML: SafeHtml;
 
-  constructor(@Optional() private parent: FaLayersComponent, private sanitizer: DomSanitizer) {
+  constructor(@Optional() private parent: FaLayersComponent, private sanitizer: DomSanitizer, private renderer: Renderer2, private elementRef: ElementRef) {
     faWarnIfParentNotExist(this.parent, 'FaLayersComponent', this.constructor.name);
   }
 
+  ngOnInit() {
+    this.renderer.removeClass(this.elementRef.nativeElement, 'fa-layers-counter');
+  }
+
   ngOnChanges(changes: SimpleChanges) {
     if (changes) {
+      if ('position' in changes) {
+        if (changes.position.currentValue !== null) {
+          this.classes.pop();
```

Comment:
Wouldn't this behave incorrectly when user attempts to pass a custom class using the `classes` input. E.g. `<fa-layers-counter [classes]="['my-custom-class']" position="top-right"></fa-layers-counter>` will not have `my-custom-class`. 

It is also pretty dangerous to mutate the array passed in as an input as it may have unintended effects on repeatable render passes, I guess that's why you had to add that `.pop()` call at the first place.

IMO it would be safer and cleaner to build a new classes array with all relevant classes in the `buildParams` method. Something like below. WDYT?

```ts
const classes = [];
if (this.classes != null) {
  classes.push(...this.classes);
}
if (this.position != null) {
  classes.push(`fa-layers-${changes.position.currentValue}`);
}
return {
      title: this.title,
      classes: classes,
      styles: this.styles,
    };
```

Reply:
You're right. Thank you for the feedback!